page_delete_test.go

```diff
@@ -0,0 +1,135 @@
+package eventstream
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var pageDeleteTestSince = time.Now().UTC()
+
+const pageDeleteTestExecURL = "/page-delete-exec"
+const pageDeleteTestSubURL = "/page-delete-sub"
+
+func createPageDeleteServer(t *testing.T, since *time.Time) (http.Handler, error) {
+	router := http.NewServeMux()
+	stubs, err := readStub("page-delete.json")
+
+	if err != nil {
+		return router, err
+	}
+
+	router.HandleFunc(pageDeleteTestExecURL, func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, since.Format(time.RFC3339), r.URL.Query().Get("since"))
+
+		f := w.(http.Flusher)
+
+		for _, stub := range stubs {
+			w.Write(stub)
+			f.Flush()
+		}
+	})
+
+	router.HandleFunc(pageDeleteTestSubURL, func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, since.Format(time.RFC3339), r.URL.Query().Get("since"))
+
+		f := w.(http.Flusher)
+
+		for _, stub := range stubs {
+			w.Write(stub)
+			f.Flush()
+		}
+	})
+
+	return router, nil
+}
+
+func TestPageDeleteExec(t *testing.T) {
+	router, err := createPageDeleteServer(t, &pageDeleteTestSince)
+	assert.Nil(t, err)
+
+	srv := httptest.NewServer(router)
+	defer srv.Close()
+
+	client := NewBuilder().
+		URL(srv.URL).
+		Options(&Options{
+			PageDeleteURL: pageDeleteTestExecURL,
+		}).
+		Build()
+
+	type msg struct {
```

Comment:
This is no the right place to declare structs, please make declarations on top of the file.